### PR TITLE
fix resize luks volume

### PR DESCRIPTION
**Pull request by outscale-hmi**
see https://github.com/outscale/osc-bsu-csi-driver/pull/837

### CODE REVIEW

File: pkg/driver/luks_util.go [hunk 1]
```diff
@@ -86,10 +86,22 @@ func IsLuksMapping(exec k8sExec.Interface, devicePath string) (bool, string, err
 	return false, "", nil
 }
 
-func LuksResize(exec k8sExec.Interface, deviceName string) error {
+func LuksResize(exec k8sExec.Interface, deviceName string, passphrase string) error {
 	cryptsetupArgs := []string{"--batch-mode", "resize", deviceName}
```
Comment (outscale-rce) [inline pkg/driver/luks_util.go:90]: // Define arguments for cryptsetup
cryptsetupArgs := []string{"--batch-mode", "resize", deviceName}

File: pkg/driver/luks_util.go [hunk 2]
```diff
@@ -86,10 +86,22 @@ func IsLuksMapping(exec k8sExec.Interface, devicePath string) (bool, string, err
...
 
-	return exec.Command("cryptsetup", cryptsetupArgs...).Run()
+	// Créer la commande cryptsetup
```
Comment (outscale-rce) [inline pkg/driver/luks_util.go:92]: // Create the resize command

File: pkg/driver/luks_util.go [hunk 3]
```diff
@@ -86,10 +86,22 @@ func IsLuksMapping(exec k8sExec.Interface, devicePath string) (bool, string, err
...
+	resizeCmd := exec.Command("cryptsetup", cryptsetupArgs...)
+
+	// Passer la passphrase via stdin
```
Comment (outscale-rce) [inline pkg/driver/luks_util.go:95]: // Set passphrase as stdin input

File: pkg/driver/luks_util.go [hunk 4]
```diff
@@ -86,10 +86,22 @@ func IsLuksMapping(exec k8sExec.Interface, devicePath string) (bool, string, err
...
+	passwordReader := strings.NewReader(passphrase)
+	resizeCmd.SetStdin(passwordReader)
+
+	// Exécuter la commande et gérer les erreurs
```
Comment (outscale-rce) [inline pkg/driver/luks_util.go:99]: // Run the command and handle errors

File: pkg/driver/luks_util.go [hunk 5]
```diff
@@ -86,10 +86,22 @@ func IsLuksMapping(exec k8sExec.Interface, devicePath string) (bool, string, err
...
+	if out, err := resizeCmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("err: %v, output: %v", err, string(out))
```
Comment (outscale-rce) [inline pkg/driver/luks_util.go:101]: return fmt.Errorf("failed to resize LUKS volume on %s: %v, output: %s", deviceName, err, string(out)

**Review by outscale-toa — APPROVED** [review]

**Pull request merged**